### DISCUSSION TITLE
How to display default array of nodes without postion ?

### DISCUSSION THREAD

**Discussion by rahul-yr** [xyflow/xyflow#3302] [Q&A]
I have default array of nodes without position. How to calculate the positions and display it

**Comment by moklick** [comment]
React Flow has no built-in layouting, but we have a guide with recommendations: https://reactflow.dev/docs/guides/layouting/

**Reply by rahul-yr** [reply]
I tried those layouts, only issue is when the nodes has dynamic content it has collisions. As well as rebalancing the whole tree needs calculation
